### PR TITLE
Adds a unit test to check for any missing initial icon states

### LINKED ISSUE #5282: Spawn with invisible hard hat
If you choose the hard hat selection, you will spawn with an extra, invisible, always equipped hard hat. I have had it happen multiple times as Engineering Apprentice, not sure if it happens on other engineer roles.

### LINKED ISSUE #7840: pAI custom skins are not showing up
After trying to select the custom skin for my pAI, my icon disappeared entirely.  Further, after that, any other chassis I attempted to select was also invisible.  I was only able to reappear by folding back up into a tablet... but that just spawned a tablet and let me continue to walk around for a bit, invisible.

### PR DESCRIPTION
**Pull request by mikomyazaki**
Local dev gets tired of fixing invisible item bugs. Lets see if this works.

Removes some items where no icons could be found for them, but they are all unmapped, unused, unimportant and old. No one will notice.

Fixes #7840
Fixes #5282

All PRs will need to re-run travis with this unit test included, if they touch any icon, icon_state vars or dmis.

Items that use `update_icon` to modify an `initial(icon_state)` will now need to rely on another var to hold the initial icon state, e.g. `var/modify_icon` rather than having a non-existant truncated icon_state name in `icon_state`. `icon_state` can be set to `null` in these cases.

### CODE REVIEW

**Comment by mikomyazaki** [comment]
1344 errors to fix. Help.

**Comment by mikomyazaki** [comment]
!bugfix

**Comment by mikomyazaki** [comment]
!review

File: code/unit_tests/icon_tests.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			var/list/valid_states = icon_states(atom_icon)
+			if(!(atom_icon_state in valid_states))
+				failed = TRUE
+				log_unit_test("[ascii_red]--------------- [atom_type] - Icon state [atom_icon_state] was not found in [atom_icon].")
+
+		if(failed)
+			failed_icons += atom_type
+
+	if(LAZYLEN(failed_icons))
+		fail("One or more atom initial icons were not found.")
```
Comment (Sindorman) [inline code/unit_tests/icon_tests.dm]:
```suggestion
		fail("[LAZYLEN(failed_icons)] atom initial icons were not found.")
```

**Pull request merged**